src/fsharp/LexFilter.fs

```diff
@@ -683,7 +711,8 @@ type LexFilterImpl (lightStatus: LightSyntaxStatus, compilingFsLib, lexer, lexbu
     //
     // Undentation rules
     //--------------------------------------------------------------------------
-
+    
+    let (|RelaxWhitespace2|_|) x = if lexbuf.SupportsFeature LanguageFeature.RelaxWhitespace2 then Some x else None
```

Comment:
I'm concerned the use of an active pattern will mean the pattern matching below will compile less efficiently.  I think perhaps we should do a performance test for lexing for this

Reply:
I am updating this to use `when` conditions rather than partial active patterns